In [1]:
import pandas as pd

In [2]:
pd.read_json('train.json')

,id,cuisine,ingredients
0,10259,greek,"[romaine lettuce, black olives, grape tomatoes..."
1,25693,southern_us,"[plain flour, ground pepper, salt, tomatoes, g..."
2,20130,filipino,"[eggs, pepper, salt, mayonaise, cooking oil, g..."
3,22213,indian,"[water, vegetable oil, wheat, salt]"
4,13162,indian,"[black pepper, shallots, cornflour, cayenne pe..."
...,...,...,...
39769,29109,irish,"[light brown sugar, granulated sugar, butter, ..."
39770,11462,italian,"[KRAFT Zesty Italian Dressing, purple onion, b..."
39771,2238,irish,"[eggs, citrus fruit, raisins, sourdough starte..."
39772,41882,chinese,"[boneless chicken skinless thigh, minced garli..."


Read Json also has multiple attributes. Which can be used as per need. 


https://pandas.pydata.org/docs/reference/api/pandas.read_json.html

In [4]:
pd.read_json('https://api.exchangerate-api.com/v4/latest/NPR')

,provider,WARNING_UPGRADE_TO_V6,terms,base,date,time_last_updated,rates
NPR,https://www.exchangerate-api.com,https://www.exchangerate-api.com/docs/free,https://www.exchangerate-api.com/terms,NPR,2026-06-30,1782777601,1.0000
AED,https://www.exchangerate-api.com,https://www.exchangerate-api.com/docs/free,https://www.exchangerate-api.com/terms,NPR,2026-06-30,1782777601,0.0243
AFN,https://www.exchangerate-api.com,https://www.exchangerate-api.com/docs/free,https://www.exchangerate-api.com/terms,NPR,2026-06-30,1782777601,0.4250
ALL,https://www.exchangerate-api.com,https://www.exchangerate-api.com/docs/free,https://www.exchangerate-api.com/terms,NPR,2026-06-30,1782777601,0.5470
AMD,https://www.exchangerate-api.com,https://www.exchangerate-api.com/docs/free,https://www.exchangerate-api.com/terms,NPR,2026-06-30,1782777601,2.4400
...,...,...,...,...,...,...,...
YER,https://www.exchangerate-api.com,https://www.exchangerate-api.com/docs/free,https://www.exchangerate-api.com/terms,NPR,2026-06-30,1782777601,1.5800
ZAR,https://www.exchangerate-api.com,https://www.exchangerate-api.com/docs/free,https://www.exchangerate-api.com/terms,NPR,2026-06-30,1782777601,0.1090
ZMW,https://www.exchangerate-api.com,https://www.exchangerate-api.com/docs/free,https://www.exchangerate-api.com/terms,NPR,2026-06-30,1782777601,0.1190
ZWG,https://www.exchangerate-api.com,https://www.exchangerate-api.com/docs/free,https://www.exchangerate-api.com/terms,NPR,2026-06-30,1782777601,0.1770


### Working with SQL 

we need to install the library 

In [1]:
!pip install mysql.connector

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.9/11.9 MB 3.1 MB/s  0:00:04 eta 0:00:01
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Created wheel for mysql.connector: filename=mysql_connector-2.2.9-cp314-cp314-linux_x86_64.whl size=248041 sha256=a4036cc3e768e0b3ef3b3c0bc2e70876584052bcceb2b13e8f943e4911e19a9d
  Stored in directory: /home/foolmann/.cache/pip/wheels/a2/5a/60/e03682d488f501428666827b000cfa0da00163afcd74ce2c14
Successfully built mysql.connector


In [ ]:
import mysql.connector 

In [23]:
import mysql.connector

try:
    conn = mysql.connector.connect(
        host="localhost",
        user='root',
        password='root',
        database='world'
    )
    if conn.is_connected():
        print("Success! Connected to the database.")
        conn.close()
except mysql.connector.Error as err:
    print(f"The connection failed due to this error: {err}")

The connection failed due to this error: 2003: Can't connect to MySQL server on 'localhost:3306' (111 Connection refused)


This will not work as I am using wsl and mysql is running in windows 

Yes, it will work, but "localhost" will not work in this setup. [1] 
When your Python code runs inside WSL, localhost refers to the WSL Linux environment itself, not your Windows host. Because your MySQL server is running on Windows, WSL cannot find it at that address. [2, 3, 4, 5] 
Follow these quick steps to make them talk to each other:
## 1. Get your Windows IP Address inside WSL
WSL automatically maps your Windows host IP. You can find it by running this command inside your WSL terminal: [6, 7] 

ip route show | grep default | awk '{print $3}'

(This will output an IP address, usually something like 172.25.80.1 or 192.168.x.x).
## 2. Update your Python Code
Replace localhost with the IP address you found in the step above:

conn = mysql.connector.connect(
    host="172.25.80.1",  # Replace with the exact IP from step 1
    user='root',
    password='root',
    database='world'
)

## 3. Allow Windows MySQL to Accept Remote Connections
By default, Windows MySQL only listens to requests coming from Windows itself. You need to tell it to listen to WSL. [8] 

   1. Open your MySQL Workbench or command line on Windows.
   2. Run this SQL command to see your root user restrictions:
   
   SELECT user, host FROM mysql.user WHERE user='root';
   
   3. If the host says localhost or 127.0.0.1, your Python code will still get blocked. You must allow connections from any host (%) by running:
   
   CREATE USER 'root'@'%' IDENTIFIED BY 'root';GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' WITH GRANT OPTION;
   FLUSH PRIVILEGES;
   
   [9, 10, 11] 

## 4. Check Windows Firewall (If it still fails)
Windows Firewall might block WSL traffic on port 3306.

* Open Windows Defender Firewall on your PC.
* Click Advanced Settings -> Inbound Rules -> New Rule.
* Choose Port -> TCP -> Specific local ports: 3306.
* Select Allow the connection and save it as "MySQL WSL". [12, 13] 

If you try this, let me know:

* What IP address did the WSL command give you?
* Did changing the host string output a different error (like an Access Denied error)?



So getting the actual ip address and allowing the remote connection and all user with root 

In [29]:
conn = mysql.connector.connect(host="172.24.208.1",user='root',password='root',database='world')
conn